index.js

```diff
@@ -286,6 +286,91 @@ function createQuestionnaireTemplateHelper({
         return true;
     }
 
+    // 6 - do all sections have a theme
+
+    function ensureAllSectionsHaveThemes() {
+        const errors = Object.keys(sections).reduce((acc, sectionId) => {
+            if (sectionId !== 'system' && sectionId !== 'p--check-your-answers') {
+                const sectionSchema = sections[sectionId].schema;
+
+                // if schema includes allOf && allOf.meta
+                //      allOf > meta > classification > theme must exist
+                // ELSE
+                // if schema.properties && any properties have a 'type' key
+                //      property > meta > classification > theme must exist
+                // ELSE
+                // Error
+
+                if ('allOf' in sectionSchema) {
+                    sectionSchema.allOf.forEach(property => {
+                        if ('title' in property) {
+                            if (!('meta' in property)) {
+                                acc.push({
+                                    type: 'SectionSchemaFailed',
+                                    source: `/sections/${sectionId}/schema`,
+                                    description: 'Schema does not contain meta data'
+                                });
+                            } else if (!('classifications' in property.meta)) {
+                                acc.push({
+                                    type: 'SectionSchemaFailed',
+                                    source: `/sections/${sectionId}/schema`,
+                                    description: 'Schema does not contain classifications data'
+                                });
+                            } else if (!('theme' in property.meta.classifications)) {
+                                acc.push({
+                                    type: 'SectionSchemaFailed',
+                                    source: `/sections/${sectionId}/schema`,
+                                    description: 'Schema does not contain theme data'
+                                });
+                            }
+                        }
+                    });
+                }
+
+                if ('properties' in sectionSchema) {
+                    Object.keys(sectionSchema.properties).forEach(property => {
+                        if ('type' in sectionSchema.properties[property]) {
+                            if (!('meta' in sectionSchema.properties[property])) {
```

Comment:
there is a bit of duplicate logic here - would it be worth combining/reducing this at all? Or is this by design?

Reply:
I agree, my initial attempt was to extract it into a function that validates just the meta object but I think considering the way this function is structured in the middle of an `array.reduce` it's more readable this way.